src/app/pages/components/calendar/calendar.component.ts

```diff
@@ -0,0 +1,145 @@
+import {Component, ElementRef, Input, OnInit, ViewEncapsulation} from '@angular/core';
+import * as $ from 'jquery';
+import 'fullcalendar';
+import {Options} from 'fullcalendar';
+import * as moment from 'moment';
+
+@Component({
+    template: '<div></div>',
+    selector: 'ngx-calendar',
+    styleUrls: ['./calendar.component.css'],
+    encapsulation: ViewEncapsulation.None,
+})
+export class CalendarComponent implements OnInit {
+
+    @Input() options: Options;
+    text: string;
+    calendarInitiated: boolean;
+    calendarOptions: Object;
+    todayDate: moment.Moment;
+    private YM: string;
```

Comment:
Not sure what `YM` stands for, is there a better name?
Also, why uppercase on the private variables in here?